### DISCUSSION TITLE
Negative right margin on `Row` with horizontal gutter causes overflow

### DISCUSSION THREAD

**Discussion by dgh328** [ant-design/ant-design#41916] [Q&A]
I have not dug deep into the issue so I may not be articulating it clearly. That being said:

When I use a gutter of, say, 16, the resulting `<Row/>` container has a left and right margin of -8px (to account for the 8px padding each item has).

That works fine for most scenarios. But, in the case of my website this adds a weird spacing to the right of the viewport. My specific website has a container div that has margin-left and margin-right set to `auto` so that it can be centered, with a width of 1200px and a max-width of 100% so that it does not go over the screen on mobile. You can see it here: https://stage.zarobky.pl (visit from mobile via dev console). You can see the green background to the right caused by this.

I suspect this is why MUI's grid does something a bit different: while it also has a -8px left margin, instead of having a -8px right margin it sets the width of the container to `calc(100% + 16px)`. The resulting site does not go overboard.

I hope the above is understandable.

**Comment by MaxDonchenko** [comment]
For whoever lands in this page (as I did today) here's a workaround that will most likely work for you:
https://github.com/ant-design/ant-design/issues/10144#issuecomment-516960601
